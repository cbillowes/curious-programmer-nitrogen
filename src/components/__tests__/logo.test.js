import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Constants from "./const"
import Logo from '../logo'

Enzyme.configure({ adapter: new Adapter() })

describe('Logo', () => {

  it('should render', () => {
    const wrapper = shallow(
      <Logo />
    )

    const rendered = wrapper.html()
    expect(rendered.startsWith(`<div `)).toBe(true)
  })

  it('should link to the home page', () => {
    const wrapper = shallow(
      <Logo />
    )

    const rendered = wrapper.html()
    expect(rendered.indexOf(`href="/"`)).toBeGreaterThan(-1)
  })

  it('should render the logo content', () => {
    const wrapper = shallow(
      <Logo />
    )

    const rendered = wrapper.html()
    expect(rendered.indexOf(`${Constants.LOGO}`)).toBeGreaterThan(-1)
  })

})
