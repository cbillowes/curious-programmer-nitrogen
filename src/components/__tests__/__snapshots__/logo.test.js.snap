// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should render 1`] = `
<div
  className="logo"
>
   
  <a
    aria-current="page"
    className="full"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
    style={Object {}}
    title="Curious Programmer"
  >
    <strong>
      {
    </strong>
     :curious 
    <strong>
      "programmer"
    </strong>
     
    <strong>
      }
    </strong>
  </a>
   
   
  <a
    aria-current="page"
    className="partial"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
    style={Object {}}
    title="Curious Programmer"
  >
    <strong>
      { }
    </strong>
  </a>
   
</div>
`;
